commercetools/resource_category.go

```diff
@@ -0,0 +1,411 @@
+		newMetaDescription := commercetools.LocalizedString(expandStringMap(d.Get("meta_description").(map[string]interface{})))
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategorySetMetaDescriptionAction{MetaDescription: &newMetaDescription})
+	}
+
+	if d.HasChange("meta_keywords") {
+		newMetaKeywords := commercetools.LocalizedString(expandStringMap(d.Get("meta_keywords").(map[string]interface{})))
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategorySetMetaKeywordsAction{MetaKeywords: &newMetaKeywords})
+	}
+
+	if d.HasChange("assets") {
+		assets := resourceCategoryGetAssets(d)
+		for _, asset := range assets {
+			input.Actions = append(
+				input.Actions,
+				&commercetools.CategoryChangeAssetNameAction{Name: asset.Name, AssetKey: asset.Key},
+				&commercetools.CategorySetAssetDescriptionAction{Description: asset.Description, AssetKey: asset.Key},
+				&commercetools.CategorySetAssetSourcesAction{Sources: asset.Sources, AssetKey: asset.Key},
+			)
+			if len(asset.Tags) > 0 {
+				input.Actions = append(
+					input.Actions,
+					&commercetools.CategorySetAssetTagsAction{Tags: asset.Tags, AssetKey: asset.Key},
+				)
+			}
+		}
+	}
+
+	log.Printf(
+		"[DEBUG] Will perform update operation with the following actions:\n%s",
+		stringFormatActions(input.Actions))
+
+	_, err = client.CategoryUpdateWithID(context.Background(), input)
+	if err != nil {
+		if ctErr, ok := err.(commercetools.ErrorResponse); ok {
+			log.Printf("[DEBUG] %v: %v", ctErr, stringFormatErrorExtras(ctErr))
+		}
+		return err
+	}
+
+	return resourceCategoryRead(d, m)
+}
+
+func resourceCategoryDelete(d *schema.ResourceData, m interface{}) error {
+	client := getClient(m)
+
+	version := d.Get("version").(int)
+	_, err := client.CategoryDeleteWithID(context.Background(), d.Id(), version)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceCategoryGetAssets(d *schema.ResourceData) []commercetools.AssetDraft {
+	input := d.Get("assets").([]interface{})
+	var result []commercetools.AssetDraft
+
+	for _, raw := range input {
+		i := raw.(map[string]interface{})
+
+		name := commercetools.LocalizedString(expandStringMap(i["name"].(map[string]interface{})))
+		description := commercetools.LocalizedString(expandStringMap(i["description"].(map[string]interface{})))
+		sources := resourceCategoryGetAssetSources(i)
+
+		result = append(result, commercetools.AssetDraft{
+			Key:         i["key"].(string),
+			Name:        &name,
+			Description: &description,
+			Sources:     sources,
+		})
+	}
+
+	return result
+}
+
+func resourceCategoryGetAssetSources(i map[string]interface{}) []commercetools.AssetSource {
+	var sources []commercetools.AssetSource
+	for _, item := range i["sources"].([]interface{}) {
+		s := item.(map[string]interface{})
+
+		source := commercetools.AssetSource{
+			URI:         s["uri"].(string),
+			Key:         s["key"].(string),
+			ContentType: s["content_type"].(string),
```

Comment:
content_type and key props  are optional and might be nil ( casting nil to string will cause panic ).

I see you have a test where you create a category with an asset that has no content type .
if it works thas weird.